File: abc163/b/v1.scala
// AC

object Main extends App {
  val sc = new java.util.Scanner(System.in);
  val n, m = sc.nextInt();
  val as = IndexedSeq.fill(m)(sc.nextInt());
  val answer = n - as.sum;
  if (answer < 0) {
    println(-1);
  } else {
    println(answer);
  }
}

File: abc173/d/v1.java
// AC

import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    var sc = new Scanner(System.in);
    int n = sc.nextInt();
    var as = new Integer[n];
    for (int i = 0; i < n; i++) {
      as[i] = sc.nextInt();
    }
    java.util.Arrays.sort(as, java.util.Comparator.comparingInt(x -> -x));
    long answer = 0;
    for (int i = 1; i < n; i++) {
      answer += as[i / 2];
    }
    System.out.println(answer);
  }
}
